scripts/generate_mnx_shortlists.py

```diff
@@ -0,0 +1,109 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2017 Novo Nordisk Foundation Center for Biosustainability,
+# Technical University of Denmark.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# MetaNetX has been published as:
+# MetaNetX/MNXref - reconciliation of metabolites and biochemical reactions to
+# bring together genome-scale metabolic networks
+# Sébastien Moretti, Olivier Martin, T. Van Du Tran, Alan Bridge,
+# Anne Morgat and Marco Pagni
+# Nucleic Acids Research (2016), 44(D1):D523-D526
+# https://doi.org/10.1093/bioinformatics/btt036
+# https://www.metanetx.org/
+#
+
+"""Scripts to generate a reduced shortlist from MetaNetX database dumps."""
+
+import pandas as pd
+
+# Read the database dumps in the correct format: remove comment lines, and
+# provide appropriate column names.
+
+xref = pd.read_csv(
+    "chem_xref.tsv",
+    delim_whitespace=True,
+    comment='#',
+    names=['XREF', 'MNX_ID', 'Evidence', 'Description'])
+
+
+def xref_splitter(xref):
+    """
+    Return database name from a combined string like "bigg" in "bigg:pyr".
+
+    Return 'mnx' as the ID when the string in xref contains 'MNX' but does not
+    contain a colon.
+
+    Parameters
+    ----------
+    xref : pandas.Dataframe
+        The metabolic model under investigation.
+
+    """
+    if ':' in xref:
+        return xref.split(':')[0]
+    elif 'MNX' in xref and ':' not in xref:
+        return 'mnx'
+
+
```

Comment:
Would it make sense to limit the dataframe to identifiers from the shortlist before performing all these transformations?

Reply:
It is a bit tricky because the initial dataframe isn't shaped in a way that allows that but it would make the execution of the script much faster.

I've made the appropriate changes to the script!